engine/src/java/com/runrev/android/Engine.java

```diff
@@ -2205,33 +2206,98 @@ else if (resultCode == Activity.RESULT_OK)
-				{
-					t_out.write(t_buffer, 0, t_readcount);
-				}
-
+				
 				// HH-2017-01-19: [[ Bug 11313 ]]Support maximum width and height of the image
 				if(m_photo_height > 0 && m_photo_width > 0)
 				{
-					Bitmap bm = BitmapFactory.decodeByteArray(t_out.toByteArray(), 0, t_out.size());
-					Bitmap rBm = Bitmap.createScaledBitmap(bm, m_photo_width, m_photo_height, true);
-					ByteArrayOutputStream stream = new ByteArrayOutputStream();
-					rBm.compress(Bitmap.CompressFormat.PNG, 100, stream);
-					byte[] byteArray = stream.toByteArray();
-					doPhotoPickerDone(byteArray, byteArray.length);
+					Bitmap t_bitmap = BitmapFactory.decodeStream(t_in);
+					
+					// scale to required max width/height
+					float t_width = t_bitmap.getWidth();
+					float t_height = t_bitmap.getHeight();
+					
+					InputStream t_exif_in = ((LiveCodeActivity)getContext()).getContentResolver().openInputStream(t_photo_uri);
+					
+					ExifInterface t_exif = new ExifInterface(t_exif_in);
+					
+					int t_orientation = t_exif.getAttributeInt(ExifInterface.TAG_ORIENTATION,
+															   ExifInterface.ORIENTATION_NORMAL);
+					
+					/* Max width and height need to be flipped if the image orientation is
+					 * requires a 90 or 270 degree rotation */
+					int t_max_width;
+					int t_max_height;
+					switch (t_orientation)
+					{
+						case ExifInterface.ORIENTATION_TRANSPOSE:
+						case ExifInterface.ORIENTATION_ROTATE_90:
+						case ExifInterface.ORIENTATION_TRANSVERSE:
+						case ExifInterface.ORIENTATION_ROTATE_270:
+							t_max_height = m_photo_width;
+							t_max_width = m_photo_height;
+							break;
+						default:
+							t_max_width = m_photo_width;
+							t_max_height = m_photo_height;
+							break;
+					}
+					
+					float t_scale = Math.min(t_max_width / t_width, t_max_height / t_height);
+					
+					Matrix t_matrix = new Matrix();
+					t_matrix.setScale(t_scale, t_scale, 0, 0);
+					
+					switch (t_orientation)
+					{
+						case ExifInterface.ORIENTATION_FLIP_HORIZONTAL:
+							t_matrix.postScale(-1, 1);
+							break;
+						case ExifInterface.ORIENTATION_ROTATE_180:
+							t_matrix.postRotate(180);
+							break;
+						case ExifInterface.ORIENTATION_FLIP_VERTICAL:
+							t_matrix.postRotate(180);
+							t_matrix.postScale(-1, 1);
+							break;
+						case ExifInterface.ORIENTATION_TRANSPOSE:
+							t_matrix.postRotate(90);
+							t_matrix.postScale(-1, 1);
+							break;
+						case ExifInterface.ORIENTATION_ROTATE_90:
+							t_matrix.postRotate(90);
+							break;
+						case ExifInterface.ORIENTATION_TRANSVERSE:
+							t_matrix.postRotate(-90);
+							t_matrix.postScale(-1, 1);
+							break;
+						case ExifInterface.ORIENTATION_ROTATE_270:
+							t_matrix.postRotate(-90);
+							break;
+					}
+		
+					Bitmap t_scaled_bitmap = Bitmap.createBitmap (t_bitmap, 0, 0, (int)t_width, (int)t_height, t_matrix, true);
+					t_scaled_bitmap.compress(Bitmap.CompressFormat.JPEG, 100, t_out);
```

Comment:
So this uses a quality of 100% - can you try and find out what 'quality' level is used on iOS with the means we use there to get a JPEG out of a scaled bitmap?

Reply:
On iOS it uses `MCjpegquality / 100.0f`, however, `MCjpegquality` appears to have nothing setting it to anything other than 100.

Reply:
Okay so we should be passing MCjpegquality into the android method then :)

Reply:
Aha... sorry my mistake I didn't search for it correctly. It's a global prop I didn't know about! OK Will see if this can be used somehow.